docs/runtime/initializing-storage.md

```diff
@@ -0,0 +1,140 @@
+
+> The runtime developer may choose to give the field a specific name by including it in the parentheses like this
+>
+> `pub Ante get(ante) config(value_to_ante): u32;`
+>
+> If no custom name is supplied, the [snake_case](https://en.wikipedia.org/wiki/Snake_case) version of the storage field is used.
+
+For this option, and all remaining options, the deployer must use `Config<T>` when installing the module in their runtime with something like.
+
+```rust
+MyModule: my_module::{Module, Call, Storage, Event<T>, Config<T>},
+```
+
+The deployer then has two options for how to specify this value.
+
+### Enter Values in Rust Code
+
+When a deployer later includes this runtime module as part of their runtime, they may code the initial storage value directly into Rust code. If their node looks similar Substrate's node template, this code would go in `src/chain_spec.rs`. And, for our example module, would look like this.
+
+```rust
+use my_blockchain_runtime::MyModuleConfig;
+...
+GenesisConfig {
+  ...
+  template: Some(TemplateModuleConfig {
+      ante: 5,
+      min_raise: 7,
+  }),
+  ...
+}
+```
+
+> A bug in the v1.0 branch, occasionally causes a compilation error when specifying the genesis configuration this way. Adding an additional field will resolve it.
+>
+> `_genesis_phantom_data: Default::default(),`
+
+### Enter Values in Configuration File
+
+While there must always be some value specified in code, the deployer may choose to override those values via a configuration file. In Substrate-based chains a chainspec can be generated with `substrate build-spec > chainspec.json`. For our example module, the configuration would look like this.
+
+```json
+{
+  ...
+  "genesis": {
+    "runtime": {
+      ...
+      "my_module": {
+        "ante": 55,
+        "minRaise": 77
+      }
+    }
+  }
+}
+```
+
+## Calculate Individually with `build()`
+
+The third option is ideal when the runtime module developer wants to specify the field's initial value, but it is not known at compile time. For example, it may need to be calculated from other initial values that are deferred to the deployer. To choose this option the runtime developer may include `build(<closure>)` and a [closure](https://doc.rust-lang.org/stable/book/ch13-01-closures.html) that returns the desired initial value.
+
+In our example game, the minimum raise must always be twice the ante. But the ante may be configured by the module deployer.
+```rust
+decl_storage! {
+  trait Store for Module<T: Trait> as TemplateModule {
+    // Points ante'd to play the game
+    pub Ante get(ante) config(): u32;
+
+      // Minimum raise on each turn
+      pub MinRaise get(min_raise) build(|config: &GenesisConfig<T>| config.ante * 2): u32;
+  }
+}
+```
+> A bug in the v1.0 branch, occasionally causes a compilation error when specifying the genesis configuration this way. Adding an additional field will resolve it.
+>
+> `_genesis_phantom_data: Default::default(),`
```

Comment:
I'm not sure this comment here makes sense no ?

Reply:
The context here is
```rust
decl_storage! {
  trait Store for Module<T: Trait> as TemplateModule {
    pub Ante get(ante) config(): u32;
    pub MinRaise get(min_raise) build(|config: &GenesisConfig<T>| config.ante * 2): u32;
  }
}
```
Where `ante` goes in the genesis config, and `min_raise` is calculated with build. My experience testing this was that I needed `chain_spec.rs` to look like
```rust
template: Some(TemplateModuleConfig {
      ante: 5,
      // Woulkdn't compile without this.
      _genesis_phantom_data: Default::default(),
  }),
```

Reply:
oh ok like because MinRaise is no longer into config it now needs a phantom_data, ok make sense then it worth keeping it